### ISSUE TITLE
KeyVault List includes resources other than KeyVaults

### ISSUE THREAD

**Issue by marstr** [Azure/azure-rest-api-specs#1464]
In Azure/azure-sdk-for-go#696 @chessman says:
> `ListPreparer` uses path `"/subscriptions/{subscriptionId}/resources"`. But it should be something like `"/subscriptions/{subscriptionId}/providers/Microsoft.KeyVault/vaults"`.

I built out a repro for the Go SDK that confirms this behavior and can be found here: https://github.com/marstr/issue696

**Comment by ravbhatnagar** [comment]
Not an ARM issue. This needs to be fixed by KeyVault team

**Comment by marstr** [comment]
@ravbhatnagar, why would that be? This doesn't have to do with interacting with an individual KeyVault, but rather managing instances of them through ARM.

**Comment by marstr** [comment]
@mcardosos was just chatting with me and says she has some more context here. Some of what she was saying may back up the assertion from @ravbhatnagar that this is fundamentally a problem the service team needs to fix.

Regardless, I think the question of whether or not this operation should be exposed through the Swagger today in its current form should be answered. Without KeyVault service-team intervention, I'd like to throw the "deprecated" tag on this and document how we would recommend performing the same operation through the "resources" RP.

**Comment by mcardosos** [comment]
[More context](https://github.com/Azure/azure-sdk-for-go/issues/403)
TL;DR: This operation calls ARM API directly, not through the KeyVault API.
This causes the SDKs to be used in awkward ways, for example: `resourceClient.List("resourceType eq 'Microsoft.KeyVault/vaults'", "", nil)`. The path should be corrected to `"/subscriptions/{subscriptionId}/providers/Microsoft.KeyVault/vaults"`. That way we can also get rid of the `$filter` parameter, and hopefully, the custom `api-version` too. 

I would like this operation to be defined like this...
``` json
"/subscriptions/{subscriptionId}/providers/Microsoft.KeyVault/vaults": {
      "get": {
        "tags": [
          "Vaults"
        ],
        "operationId": "Vaults_ListBySubscription",
        "description": "The List operation gets information about the vaults associated with the subscription.",
        "parameters": [
          {
            "name": "$top",
            "in": "query",
            "type": "integer",
            "format": "int32",
            "description": "Maximum number of results to return."
          },
          {
            "$ref": "#/parameters/ApiVersionParameter"
          },
          {
            "$ref": "#/parameters/SubscriptionIdParameter"
          }
        ],
        "responses": {
          "200": {
            "description": "Get information about all key vaults in the subscription.",
            "schema": {
              "$ref": "#/definitions/ResourceListResult"
            }
          }
        },
        "x-ms-pageable": {
          "nextLinkName": "nextLink"
        }
      }

**Comment by jlichwa** [comment]
Currently by design Key Vault does not support efficient to list resources outside the region. It is by design that ARM rest api is used instead to list resources for subscriptions across regions. 
#please-close as as designed.
